### PR TITLE
refactor: error handling in connect.Connect

### PR DESCRIPTION
**Pull request by markfeinstein**
Restructure the code around the connect.Connect method to return errors up the stack for centralized handling. 

fixes: #31

### CODE REVIEW

File: connect/connect.go [hunk 1]
```diff
@@ -1,17 +1,26 @@
...
-	session := session.Determine(choice)
-	zoxide.Add(session.Path)
-	tmux.Connect(tmux.TmuxSession{
-		Name: session.Name,
-		Path: session.Path,
-	}, alwaysSwitch, command)
-	return nil
+	session, err := session.Determine(choice)
+	if err != nil {
+		return fmt.Errorf("unable to connect to %q: %w", choice, err)
```
[the remark this answers is not in the document]
Comment (markfeinstein) [reply connect/connect.go:14]: Not sure if I am understanding your question. 

This `fmt` statement decorates an error that is being returned to the calling function. It doesn't directly display anything to the user. We can have additional error handling / formatting closer to the command execution code, but this part is intended to add context about what the application was trying to accomplish while the error occurred.

File: session/determine.go [hunk 2]
```diff
@@ -1,22 +1,30 @@
...
+	"fmt"
 	"log"
 )
 
-func Determine(choice string) Session {
+func Determine(choice string) (s Session, err error) {
 	path, err := DeterminePath(choice)
 	if err != nil {
-		log.Fatal("Couldn't determine the session path", err)
+		return s, fmt.Errorf(
```
[the remark this answers is not in the document]
Comment (markfeinstein) [reply session/determine.go:11]: Best practices can be somewhat subjective, but in my opion errors should be returned up the call path and displaying them to the user should be handled centrally.

File: tmux/tmux.go [hunk 3]
```diff
@@ -85,7 +87,7 @@ func Connect(s TmuxSession, alwaysSwitch bool, command string) error {
 	if !isSession {
 		_, err := NewSession(s)
 		if err != nil {
-			log.Fatal(err)
+			fmt.Errorf("unable to connect to tmux session %q: %w", s.Name, err)
```
[the remark this answers is not in the document]
Comment (markfeinstein) [reply tmux/tmux.go:90]: `fmt.Errorf` formats an error but it doesn't print it, it also does not affect the parrent process or the appications exit code.

**Comment by markfeinstein** [comment]
@joshmedeski We can reduce the error handling here as much as you'd like, but fwiw Go's error handling tends to look rather repetative and rather verbose. While it is possible to use a `panic` to terminate a running process and return an error that is typically discouraged in favor of handling errors where they happen / where they are returned.

File: session/determine.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshmedeski) [reply session/determine.go:11]: Okay, I understand, thanks for clarifying

File: connect/connect.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshmedeski) [reply connect/connect.go:14]: I guess I was thinking we could just return error but your explanation makes sense, thanks.

**Pull request merged**